drivers/i2c/i2c_ll_stm32.c

```diff
@@ -391,6 +389,11 @@ static int i2c_stm32_init(const struct device *dev)
 	cfg->irq_config_func(dev);
 #endif
 
+#ifdef I2C_STM32_V2_DMA
+	k_sem_init(&data->dma_rx_sem, 1, K_SEM_MAX_LIMIT);
```

Comment:
What is the rationale behind having lock:s for DMA?
Should'nt  the "bus_mutex" in  struct i2c_stm32_data be enogh to protect both I2C
and DMA:s from mutial access?

Reply:
it's used to protect the data block config from being updated before the transfer was complete. I had an oddity during testing and this was my working solution

Reply:
Seems to operate cleanly without the semaphores so I have removed them